en/3-transaction/types.md

```diff
@@ -0,0 +1,284 @@
+
+| Field         | Type  | Description           |
+|---------------|-------|-----------------------|
+| `descriptors` | array | State descriptor list |
+
+#### Descriptors
+A `descriptor` contains voting information. It can be used to apply as a consensus node or to vote for a consensus node candidate.
+
+| Field   | Type   | Description                          |
+|---------|--------|--------------------------------------|
+| `type`  | uint8  | State descriptor type                |
+| `key`   | byte   | Contract script hash or public key   |
+| `field` | string | State field to update                |
+| `value` | byte   | Updated value for the state field    |
+
+#### Type
+The state descriptor type can be either `0x40` for `Account` or `0x48` for `Validator`. If this type is set to account then the descriptor can be used to update votes. If this type is set to validator then it is used to update the registration state as a consensus node candidate.
+
+#### Key
+The key is either the 20 byte script hash of the voting contract if `Type` is `Account` or the 33 byte public key of the validator if `Type` is `Validator`.
+
+#### Field
+The state field to update. This is either `Registered` to set validator request or `Votes` to set account votes.
+
+#### Value
+Updated value for the state field. If a validator wants to register to be a consensus node candidate then he can set the `Registered` field to the value `True` or cancel a registration by setting `Registered` to `False`. If an account wants to vote for consensus nodes he can set an array of public keys to vote for.
+
+Example:
+
+```json
+{
+   "txid":"0x7031bbc63b5087ed1e84a7a6f02a4b5c7a998c800862f0b5d1e745128bdd152f",
+   "size":271,
+   "type":"StateTransaction",
+   "version":0,
+   "attributes":[
+
+   ],
+   "vout":[
+
+   ],
+   "vin":[
+
+   ],
+   "sys_fee":"0",
+   "net_fee":"0",
+   "scripts":[
+      {
+         "invocation":"40f74883a23083f9a236b01aec3eb016e4611ac23cf3036addf8957460917bad4bb968cbbb93e321da368e11da77defca0342299dd8806e9953c0def13c6b348f4",
+         "verification":"21034278b37db70b6c59b1e68a34569b77c5470ed50b9a1176b0d2da1b1b601ae86cac"
+      }
+   ],
+   "descriptors":[
+      {
+         "type":"Account",
+         "key":"ce47912d51b7ed8a24a77f97f2cb6ef9a0e0bf0d",
+         "field":"Votes",
+         "value":"04024c7b7fb6c310fccf1ba33b082519d82964ea93868d676662d4a59ad548df0e7d02aaec38470f6aad0042c6e877cfd8087d2676b0f516fddd362801b9bd3936399e02ca0e27697b9c248f6f16e085fd0061e26f44da85b58ee835c110caa5ec3ba55402df48f60e8f3e01c48ff40b9b7f1310d7a8b2a193188befe1c2e3df740e895093"
+      }
+   ]
+}
+```
+
+### InvocationTransaction
+Invoke or deploy a smart contract on the blockchain. 
+
+#### Additional attributes for a InvocationTransaction:
+
+| Field    | Type  | Description                                  |
+|----------|-------|----------------------------------------------|
+| `script` | byte  | Script to be executed by the virtual machine |
```

Comment:
Fixed8 Gas not included

Reply:
You are right. I left this out on purpose because the `gas` value appears to be the exact same as the `sys_fee` value. I went through all the transactions to check and these values are the same always. Any clue why we have these two separate attributes?